pkg/roachprod/prometheus/prometheus.go

```diff
@@ -0,0 +1,487 @@
+// Copyright 2021 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package prometheus
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/roachprod/config"
+	"github.com/cockroachdb/cockroach/pkg/roachprod/install"
+	"github.com/cockroachdb/cockroach/pkg/roachprod/logger"
+	"github.com/cockroachdb/errors"
+	promv1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	"github.com/prometheus/common/model"
+	"gopkg.in/yaml.v2"
+)
+
+const workloadPrometheusPort = 2112
+
+// Client is an interface allowing queries against Prometheus.
+type Client interface {
+	Query(ctx context.Context, query string, ts time.Time) (model.Value, promv1.Warnings, error)
+}
+
+// ScrapeNode is a node to scrape from.
+type ScrapeNode struct {
+	Nodes install.Nodes
+	IPs   []string
```

Comment:
Do we need to include these IPs here? Since we already have the list of nodes, why not fetch the IPs on demand when needed? This is what we were doing in the other prometheus.go file. The need for error handling during struct initialization makes the options constructors non-chainable.

Reply:
What's so bad about the on-the-fly roachprod.IP calls? I prefer plumbing in _just_ the node and fetching things on demand since you're not having to be defensive around not having the right 1:1 mapping at every point you get one of each, and it reduces the number of types. Actually, looking at the uses, we're not even meaningfully using `Nodes` anymore, so perhaps just choose one or the other and skip all the error handling.

Reply:
Ack. Will resolve this when I rebase and understand the best way to call these helpers there.

Reply:
Done.